src/components/dialogs/input-hooks.js

```diff
@@ -819,3 +820,58 @@ export const useExpandableValues = ({
 
     return [values, field];
 };
+
+export const useSimpleTextValue = ({ defaultValue }) => {
+    const [value, setValue] = useState(defaultValue);
+
+    const handleChangeValue = useCallback((event) => {
+        setValue(event.target.value);
+    }, []);
+
+    const field = useMemo(() => {
+        return <TextField value={value} onChange={handleChangeValue} />;
+    }, [value, handleChangeValue]);
+
+    useEffect(() => setValue(defaultValue), [defaultValue]);
+
+    return [value, field];
+};
+
+export const useValidNodeName = ({ studyUuid, defaultValue }) => {
+    const intl = useIntl();
+    const [isValidName, setIsValidName] = useState(true);
+    const [error, setError] = useState();
+    const timer = useRef();
+    const [name, field] = useSimpleTextValue({
+        defaultValue,
+    });
+
+    const validName = useCallback(
+        (name) => {
+            elementExists(studyUuid, name).then((response) => {
+                if (response.status === 200) {
+                    setError(
+                        intl.formatMessage({
+                            id: 'nodeNameAlreadyUsed',
+                        })
+                    );
+                    setIsValidName(false);
+                }
+            });
+        },
+        [studyUuid, intl]
+    );
+
+    useEffect(() => {
+        if (name === '' && !timer.current) return; // initial render
+        if (name === '' && !timer.current) {
+            return;
+        }
+        clearTimeout(timer.current);
+        setIsValidName(true);
```

Comment:
by default it's valid ? it should be false by default then true if HEAD request return not found I think